activerecord/lib/active_record/railtie.rb

```diff
@@ -217,6 +217,16 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
+    initializer "active_record.postgresql_adapter_decode_dates" do
+      config.after_initialize do
+        unless config.active_record.postgresql_adapter_decode_dates
+          ActiveSupport.on_load(:active_record_postgresqladapter) do
+            self.decode_dates = false
+          end
+        end
+      end
+    end
+
```

Comment:
@ghiculescu Just a heads up: [I've flipped the default value](https://github.com/rails/rails/pull/51763/files#diff-2468c670eb10c24bd2823e42708489a336d6f21c6efc7e3c4a574166fa77bb22R132) on the adapter itself to true.

It makes this initializer a bit more awkward, but it reduces the spam of deprecation warnings when running tests and fixes [a test that was failing](https://buildkite.com/rails/rails/builds/106954#018f5a97-c2a4-40bd-b855-a72d2d80b729) (false negative) due to the deprecation warning.

Reply:
Hmm, I don't think we should do this. Because then it's a breaking change for folks who just use Active Record without Rails.

Usually we make the default in AR the non-breaking-change variant, and then we use `load_defaults` / `new_framework_defaults` to opt into the new variant.

Reply:
Maybe just do it without the deprecation.

Reply:
I see! I'll keep that use case in mind next time. 
I've flipped it back and removed the deprecation.

I'll open another PR for the deprecation if I think of something.